crates/core/src/db/relational_db.rs

```diff
@@ -151,14 +151,37 @@ impl RelationalDB {
     where
         T: durability::History<TxData = Txdata>,
     {
-        log::debug!("DATABASE: applying transaction history...");
-        // TODO: Output progress
+        log::info!("[{}] DATABASE: applying transaction history...", self.address);
+
+        // TODO: Revisit once we actually replay history suffixes, ie. starting
+        // from an offset larger than the history's min offset.
+        let max_tx_offset = history.max_tx_offset();
+        let mut last_logged_percentage = 0;
+        let progress = |tx_offset: u64| {
+            if let Some(max_tx_offset) = max_tx_offset {
+                let percentage = f64::floor((tx_offset as f64 / max_tx_offset as f64) * 100.0) as i32;
+                if percentage > last_logged_percentage && percentage % 10 == 0 {
```

Comment:
Our general UX issue with this sort of logging would be that if it takes a really long time to get to the first 10% we really don't know how long it will take to complete the reload until we see that first 10% log. I would suggest changing this so that it outputs a log every 15 seconds instead of doing a percentage based log. If its not an easy change then don't worry about it this is fine for now.

Reply:
Hm, ya, not sure if we'd want to spawn a thread just for this, or ask for the system time a couple of million times. I'll take a note that we might want to stick a `tokio::runtime::Handle` into `RelationalDB` for these kinds of things.